### PR TITLE
fix 500 when redirected to domain select after visiting request domain invitation page

### PR DESCRIPTION
**Pull request by nickpell**

### CODE REVIEW

File: corehq/apps/domain/views.py [hunk 1]
```diff
@@ -88,6 +88,7 @@
 from dimagi.utils.decorators.memoized import memoized
 from dimagi.utils.web import get_ip, json_response, get_site_domain
 from corehq.apps.users.decorators import require_can_edit_web_users
+from corehq.apps.users.models import CouchUser
```
Comment (dimagimon) [inline corehq/apps/domain/views.py]: F401 'CouchUser' imported but unused

File: corehq/apps/domain/views.py [hunk 2]
```diff
@@ -131,11 +131,16 @@ def select(request, domain_select_template='domain/select.html', do_not_redirect
...
         if domain and domain.is_active:
-            try:
-                from corehq.apps.dashboard.views import dashboard_default
-                return dashboard_default(request, last_visited_domain)
-            except Http404:
-                pass
+            if (
+                request.couch_user.is_member_of(domain) or domain.is_public
+                or (request.user.is_superuser and not domain.restrict_superusers)
+                or domain.is_snapshot
```
Comment (czue) [inline corehq/apps/domain/views.py]: was this copied from somewhere or did you just come up with this logic? (seems right, just checking and wondering if it should be abstracted out to a function)

Comment (nickpell) [inline corehq/apps/domain/views.py]: added a comment about where the logic came from.  I don't think it's possible to abstract it out though.

**Comment by nickpell** [comment]
:white_check_mark: @orangejenny

File: corehq/apps/domain/views.py (hunk 2 above)
Comment (czue) [inline corehq/apps/domain/views.py]: :+1: thanks

**Pull request merged**